### PR TITLE
feat: [#186006497] upgrade appropriate headers to h2 with h3-styling

### PR DESCRIPTION
**Pull request by seidior**
## Description

Completing the work in #6660, this pull request completes this Pivotal Tracker story by swapping out all relevant hard-coded `###` h3 headings in our content for the corresponding `<Heading level={2} styleVariant="h3">` heading components as well as extending the work in the prior request to bring `<Heading>` to more appropriate places.

This is accomplished by the addition of one field in our CMS configuration to those relevant content items:

- `summaryDescriptionMd`: This is the paragraph that appears at the top of the detail view within the My Account dashboard. As the name belies, this contains Markdown content, and any components valid in the body content—info boxes, notes, etc.—are valid in this field as well. (This field already existed for some content collections.)

Additionally, the `##` h2 headings are now overridden for all Markdown content to be h2s with h3 styling.

### Ticket

[#186006497](https://www.pivotaltracker.com/story/show/186006497)

### Approach

This pull request contains the remaining changes necessary to complete the story, including the code and the content changes. The following content collections have been changed:

- `content/src/certifications`: Cert Opps - Content
- `content/src/display-content/business-formation/nexus`: Biz Form - DBA Tasks
- `content/src/filings`: Taxes Filings - All
- `content/src/fundings`: Funding Opps - Content
- `content/src/roadmaps/license-tasks`: License Tasks (Navigator with Webflow mappings)
- `content/src/roadmaps/tasks`: Tasks - All

The following items that contain `###` h3 headings were deemed ineligible for modifications, either because an appropriate `<Heading level={2}>` existed elsewhere in the detail view for that page in the My Account dashboard, or because it's content that does not appear within the detail view in the

- `content/src/archived-fundings`: Fund Opps - Archived
- `content/src/quick-action-tasks`: Quick Action Tasks
- `content/src/licenses`: License Expiration / Renewal Events
- `content/src/display-content/contextual-information`: Contextual Information
- `content/src/display-content/post-onboarding`: Post Onboarding Content

### Steps to Test

To test this functionality, every single content item that was changed will need to be verified to ensure the heading appears as an `<h2 class="h3-styling">` rather than an `<h3>`.

### Notes

Once this pull request is approved, we will need to merge in `content-repo` **before** rebasing and merging this branch into `main`. This will prevent conflicts in the CMS content, given the large number of content items that have been updated in this pull request and the high likelihood that some of them have changed since `content-repo` was last merged into `main`.

## Code author checklist

- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

File: content/src/certifications/dvob.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 notesMd: |-
-  :::infoAlert 
+  :::infoAlert
    hello are you?
   :::
```
Comment (jphechter) [inline content/src/certifications/dvob.md:5]: [dust]
```suggestion
```

File: web/src/components/TaskElement.tsx [hunk 2]
```diff
@@ -109,6 +111,17 @@ export const TaskElement = (props: Props): ReactElement => {
         {props.children}
         <Content>{props.task.summaryDescriptionMd || ""}</Content>
 
+        {hasBodyHeading && (
+          <>
+            <div>
+              <HorizontalLine />
+            </div>
+            <Heading level={2} styleVariant="h3">
```
Comment (jphechter) [inline web/src/components/TaskElement.tsx:119]: Design thought, not a critique of your work.

I'm curious how accurate these "variants" are. It seems to me that we're overriding more frequently than utilizing the default behavior. I only see a few instances in the codebase where we're actually using variant styling with the corresponding heading level. I get why we're doing it in code, but maybe there's an opportunity for us to rethink what these levels mean. Maybe this is a dumb thought.

File: content/src/certifications/esbe.md [hunk 3]
```diff
@@ -3,23 +3,21 @@ descriptionMd: Emerging Small Business Enterprise (ESBE) certification
   facilitates competition by small businesses for New Jersey Department of
   Transportation's (NJDOT) contracts.
 urlSlug: esbe
-name: Get a Emerging Small Business Enterprise (ESBE)
+name: Get a Emerging Small Business Enterprise (ESBE) Certification
```
Comment (jphechter) [inline content/src/certifications/esbe.md:6]: Was this change intentional?

File: content/src/fundings/food-agriculture-grant.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ---
-filename: Food and Agriculture Research & Development Grant Program
+filename: food-agriculture-grant
```
Comment (jphechter) [inline content/src/fundings/food-agriculture-grant.md:2]: I've seen a couple instances where we've gone from a kebab-case to title case and vice versa. Are these changes you're making or did this happen when you pulled in content? Just wondering if there's a rhyme or reason here.

File: content/src/fundings/new_jersey_clean_energy_loans.md
Comment (jphechter) [inline content/src/fundings/new_jersey_clean_energy_loans.md:1]: I'm assuming this deletion is coming from merging content.

File: content/src/licenses/architecture.md [hunk 5]
```diff
@@ -7,21 +7,19 @@ previewType: renewal
 callToActionLink: https://newjersey.mylicense.com/eGov/Login.aspx
 callToActionText: Renew My Architect License
 ---
+
 ### Renewal Requirements
+
```
Comment (jphechter) [inline content/src/licenses/architecture.md:12]: [sand] `bodyHeading`?

File: content/src/licenses/home-contractor.md [hunk 6]
```diff
@@ -7,17 +7,17 @@ previewType: renewal
 callToActionLink: https://newjersey.mylicense.com/eGov/Login.aspx
 callToActionText: Renew My HIC Registration
 ---
+
 ### Renewal Requirements
```
Comment (jphechter) [inline content/src/licenses/home-contractor.md:11]: [pebble] `bodyHeading`?

File: content/src/fundings/food-agriculture-grant.md (hunk 4 above)
Comment (seidior) [reply content/src/fundings/food-agriculture-grant.md:2]: This fix is a result of campfire rules. If there's spaces in the filename, it creates the filename with underscores, which is not the convention in the rest of the project.

File: content/src/fundings/new_jersey_clean_energy_loans.md
Comment (seidior) [reply content/src/fundings/new_jersey_clean_energy_loans.md:1]: Yep! The file got renamed.

File: content/src/licenses/hvac-contractor.md
Comment (jphechter) [inline content/src/licenses/hvac-contractor.md:1]: [pebble] Does this need a `bodyHeading`?

File: content/src/licenses/moving-company.md [hunk 7]
```diff
@@ -7,14 +7,17 @@ previewType: renewal
 callToActionLink: https://www.njconsumeraffairs.gov/Pages/licensingboards.aspx
 callToActionText: " Renew My Public Movers and/or Warehousemen License"
 ---
-You must renew your license every year, within 30 days of the expiration date. 
+
+You must renew your license every year, within 30 days of the expiration date.
 
 ### Renewal Requirements (If You Already Have a Registered Account)
```
Comment (jphechter) [inline content/src/licenses/moving-company.md:13]: [pebble] Does this need a `bodyHeading`?

File: content/src/licenses/architecture.md (hunk 5 above)
Comment (seidior) [reply content/src/licenses/architecture.md:12]: Good catch! License data won't be updated. I think the file only changed because it added spacing.

File: content/src/licenses/hvac-contractor.md
Comment (seidior) [reply content/src/licenses/hvac-contractor.md:1]: See above about license data.

File: content/src/licenses/moving-company.md (hunk 7 above)
Comment (seidior) [reply content/src/licenses/moving-company.md:13]: See above about license data.

File: web/public/mgmt/config.yml [hunk 8]
```diff
@@ -2714,6 +2719,8 @@ collections:
       - { label: "Url Slug / Filename", name: "urlSlug", widget: "write-once-read-only" }
       - { label: "Name", name: "name", widget: "string" }
       - { label: "Extension", name: "extension", widget: "boolean", required: false }
+      - { label: "Summary Description", name: "summaryDescriptionMd", widget: "markdown", required: false }
+      - { label: "Body Heading (h2 with h3 styling)", name: "bodyHeading", widget: "string", required: false }
```
Comment (jphechter) [inline web/public/mgmt/config.yml:2723]: [pebble] What if we made this required? In theory, content may choose not to have a header, but that seems unlikely. I don't know of an instance where we have no header. This could also future-proof us from having this same heading issue again.

File: web/src/lib/types/types.ts [hunk 9]
```diff
@@ -152,8 +152,10 @@ export type Funding = {
   urlSlug: string;
   callToActionLink: string;
   callToActionText: string;
-  contentMd: string;
   descriptionMd: string;
+  summaryDescriptionMd?: string;
+  bodyHeading?: string;
```
Comment (jphechter) [inline web/src/lib/types/types.ts:157]: [sand] If we do opt to make this required in the CMS, it should be required here as well. I'm going to leave this as one comment for the other instances.

File: web/src/lib/utils/markdownReader.ts [hunk 10]
```diff
@@ -209,7 +209,8 @@ type TaskGrayMatter = {
   requiresLocation?: boolean;
   industryId?: string;
   licenseCertificationClassification?: string;
-  summaryDescriptionMd: string;
+  summaryDescriptionMd?: string;
+  bodyHeading?: string;
```
Comment (jphechter) [inline web/src/lib/utils/markdownReader.ts:213]: [sand] Same comment. If this is required in the CMS, it should be required here.

File: web/src/pages/certification/[certificationUrlSlug].tsx [hunk 11]
```diff
@@ -36,6 +42,17 @@ export const CertificationElement = (props: { certification: Certification }): R
         <div className="margin-bottom-2">
           <h1>{props.certification.name}</h1>
         </div>
+        {descriptionAndHeadingPresent ? (
```
Comment (jphechter) [inline web/src/pages/certification/[certificationUrlSlug].tsx:45]: [pebble] Doesn't this break if one is omitted? Additionally, we've been opting for the `&&` format over the ternary for conditional rendering.

File: web/src/pages/filings/[filingUrlSlug].tsx [hunk 12]
```diff
@@ -77,6 +80,17 @@ export const FilingElement = (props: {
...
+          <>
+            <Content>{description}</Content>
+            <HorizontalLine />
+            <Heading level={2} styleVariant="h3">
+              {bodyHeading}
+            </Heading>
+          </>
+        ) : (
+          <></>
+        )}
```
Comment (jphechter) [inline web/src/pages/filings/[filingUrlSlug].tsx:93]: [pebble] Same comment.

File: web/src/pages/funding/[fundingUrlSlug].tsx [hunk 13]
```diff
@@ -57,6 +62,17 @@ export const FundingElement = (props: { funding: Funding }): ReactElement => {
...
+          <>
+            <Content>{description}</Content>
+            <HorizontalLine />
+            <Heading level={2} styleVariant="h3">
+              {bodyHeading}
+            </Heading>
+          </>
+        ) : (
+          <></>
+        )}
```
Comment (jphechter) [inline web/src/pages/funding/[fundingUrlSlug].tsx:75]: [pebble] Same comment

File: content/src/display-content/business-formation/nexus/authorize-business-entity.md [hunk 14]
```diff
@@ -8,14 +8,12 @@ callToActionLink: https://www.nj.gov/treasury/revenue/pdf/2000.pdf
 callToActionText: Authorize My Business
 agencyId: nj-revenue-enterprise-services
 required: true
+summaryDescriptionMd: >-
+  You must register your business name and `authorize|certificate-formation` your
+  `business structure|business-structure-learn-more` with the State.
+bodyHeading: Application Requirements
 ---
```
Comment (adidner) [inline content/src/display-content/business-formation/nexus/authorize-business-entity.md:16]: [dust]: Would this create different page content? With the previous ones the body header was above the summaryDescription but with this one it was below in the diff.

File: content/src/display-content/business-formation/nexus/dba-resolution-foreign.md [hunk 15]
```diff
@@ -4,14 +4,12 @@ filename: dba-resolution-foreign
 displayname: dba-resolution-foreign
 name: Get a Resolution Adopting Your DBA Name
 id: DBA-resolution-foreign
+summaryDescriptionMd: >-
+  To authorize your business in New Jersey, you need to obtain a resolution adopting your Doing Business As
+  (DBA) name.
+bodyHeading: How to Get a Resolution
 ---
```
Comment (adidner) [inline content/src/display-content/business-formation/nexus/dba-resolution-foreign.md:11]: [dust]: Would this create different page content? With the previous ones the body header was above the summaryDescription but with this one it was below in the diff.

File: content/src/display-content/business-formation/nexus/form-business-entity.md [hunk 16]
```diff
@@ -8,14 +8,12 @@ callToActionLink: https://www.njportal.com/DOR/BusinessFormation/CompanyInformat
 callToActionText: Authorize My Business
 required: true
 agencyId: nj-revenue-enterprise-services
+summaryDescriptionMd: >-
+  You must register your business name and authorize your `business structure|business-structure-learn-more`
+  with the State.
+bodyHeading: Application Requirements
 ---
```
Comment (adidner) [inline content/src/display-content/business-formation/nexus/form-business-entity.md:15]: [dust]: Would this create different page content? With the previous ones the body header was above the summaryDescription but with this one it was below in the diff.

File: content/src/fundings/angel-match.md [hunk 17]
```diff
@@ -1,5 +1,5 @@
 ---
-filename: Angel Match Program
+filename: angel-match
```
Comment (adidner) [inline content/src/fundings/angel-match.md:2]: [sand]: might be good to make this field write-once-read-only if its being editable which is seems like it is.

File: content/src/fundings/clean-energy-program-combined-heat-and-power-fuel-cells.md [hunk 18]
```diff
@@ -1,5 +1,5 @@
 ---
-filename: New Jersey’s Clean Energy Program – Combined Heat & Power - Fuel Cells
+filename: Clean Energy Program - Combined Heat and Power - Fuel Cells
```
Comment (adidner) [inline content/src/fundings/clean-energy-program-combined-heat-and-power-fuel-cells.md:2]: [dust] just wanna make sure this & vs and change was inteitional and not part of something automated

File: content/src/fundings/dep-clean-construction.md [hunk 19]
```diff
@@ -25,15 +25,17 @@ sector:
...
 dueDate: ""
-descriptionMd: "New Jersey Clean Construction Program installs pollution control devices or replaces qualifying non-road construction equipment that reduces pollution. Businesses can apply to two grants: Off-Road Diesel Retrofit Grant and Off-Road Diesel Replacement Grant."
+descriptionMd: >-
+  New Jersey Clean Construction Program installs pollution control devices or replaces qualifying non-road
+  construction equipment that reduces pollution. Businesses can apply to two grants: Off-Road Diesel Retrofit
+  Grant and Off-Road Diesel Replacement Grant.
+summaryDescriptionMd: >-
+  New Jersey Clean Construction Program installs pollution control devices or replaces qualifying non-road
+  construction equipment that reduces pollution. Businesses can apply to two grants: Off-Road Diesel Retrofit
+  Grant and Off-Road Diesel Replacement Grant.
```
Comment (adidner) [inline content/src/fundings/dep-clean-construction.md:35]: [dust] interesting that these are the same in multiples places, maybe a refactor candidate to consolidate somehow?

File: content/src/fundings/food-agriculture-grant.md (hunk 4 above)
Comment (adidner) [reply content/src/fundings/food-agriculture-grant.md:2]: Definietly should make it write-once-read-only, and maybe we should create a variant that requires nospaces as well.

File: web/src/components/tasks/EinTask.tsx [hunk 20]
```diff
@@ -63,6 +68,17 @@ export const EinTask = (props: Props): ReactElement => {
...
+        <>
+          <Content>{description}</Content>
+          <HorizontalLine />
+          <Heading level={2} styleVariant="h3">
+            {bodyHeading}
+          </Heading>
+        </>
+      ) : (
+        <></>
+      )}
```
Comment (jphechter) [inline web/src/components/tasks/EinTask.tsx:81]: [pebble] Same comment

File: web/src/components/tasks/business-formation/dba/Authorization.tsx [hunk 21]
```diff
@@ -16,6 +21,17 @@ export const Authorization = (): ReactElement => {
...
+          <>
+            <Content>{description}</Content>
+            <HorizontalLine />
+            <Heading level={2} styleVariant="h3">
+              {bodyHeading}
+            </Heading>
+          </>
+        ) : (
+          <></>
+        )}
```
Comment (jphechter) [inline web/src/components/tasks/business-formation/dba/Authorization.tsx:34]: [pebble] Same comment

File: web/src/components/tasks/business-formation/dba/DbaResolution.tsx [hunk 22]
```diff
@@ -1,12 +1,29 @@
...
+        <>
+          <Content>{description}</Content>
+          <HorizontalLine />
+          <Heading level={2} styleVariant="h3">
+            {bodyHeading}
+          </Heading>
+        </>
+      ) : (
+        <></>
+      )}
```
Comment (jphechter) [inline web/src/components/tasks/business-formation/dba/DbaResolution.tsx:26]: [pebble] Same comment

File: content/src/roadmaps/license-tasks/daycare-license.md [hunk 23]
```diff
@@ -16,13 +16,11 @@ callToActionText: Apply for My License
 webflowId: 5f7728f5e8bf507c5e317899
 licenseCertificationClassification: LICENSE
 summaryDescriptionMd: >
-  If you have more than six children under your care you must run your child care services outside of your home and will need a child care center license.
+  If you have more than six children under your care you must run your child care services outside of your
+  home and will need a child care center license.
+bodyHeading: Eligibility Criteria
```
Comment (adidner) [inline content/src/roadmaps/license-tasks/daycare-license.md:21]: [dust]: an update for consistency? gasp

File: content/src/roadmaps/tasks/liquor-license-availability.md [hunk 24]
```diff
@@ -8,13 +8,14 @@ id: liquor-license-availability
...
 callToActionText: Visit My County Clerk's Website
 ---
+
 Every local government has a unique process and cost associated with obtaining a liquor license. Contact your local clerk's office to verify the cost and availability of a liquor license in your area.
 
-:::note 
- This screen guides your next steps. It does not submit your information for registration purposes.
+:::note
+This screen guides your next steps. It does not submit your information for registration purposes.
 :::
```
Comment (adidner) [inline content/src/roadmaps/tasks/liquor-license-availability.md:16]: Wouldn't this be a summaryMD and the ### Actions to Take be a bodyHeading?

File: content/src/roadmaps/tasks/town-mercantile-license-liquor.md [hunk 25]
```diff
@@ -7,16 +7,22 @@ name: Get Your Liquor and Mercantile License
...
-- - -
+  This screen guides your next steps. It does not submit your information for registration purposes.
+
+  :::
+bodyHeading:
+---
+
+---
 
 ### Actions to Take
```
Comment (adidner) [inline content/src/roadmaps/tasks/town-mercantile-license-liquor.md:27]: What now? Shouldn't the body heading be Actions to Take?

File: content/src/roadmaps/tasks/trucking-parking.md [hunk 26]
```diff
@@ -4,11 +4,14 @@ urlSlug: trucking-parking
...
+summaryDescriptionMd: >-
+  Check with the zoning office of any location(s) where you’re interested in parking your commercial vehicle
+  for overnight storage.
 
-Check with the zoning office of any location(s) where you’re interested in parking your commercial vehicle for overnight storage.
 
-**_Before parking in a residential neighborhood, be sure to check with your local government's zoning office._**
+  **_Before parking in a residential neighborhood, be sure to check with your local government's zoning
+  office._**
+---
```
Comment (adidner) [inline content/src/roadmaps/tasks/trucking-parking.md:15]: Is it right to have back to back 

---
---

File: content/src/fundings/clean-energy-program-combined-heat-and-power-fuel-cells.md (hunk 18 above)
Comment (seidior) [reply content/src/fundings/clean-energy-program-combined-heat-and-power-fuel-cells.md:2]: Yep, intentional!

File: content/src/fundings/dep-clean-construction.md (hunk 19 above)
Comment (seidior) [reply content/src/fundings/dep-clean-construction.md:35]: This may have something to do with the Webflow syncing and the fact the body doesn't get rendered on Webflow and the summaryDescriptionMd doesn't get rendered in Navigator?

File: web/src/pages/certification/[certificationUrlSlug].tsx (hunk 11 above)
Comment (seidior) [reply web/src/pages/certification/[certificationUrlSlug].tsx]: Let's offline this conversation, if you don't mind!

File: content/src/fundings/angel-match.md (hunk 17 above)
Comment (seidior) [reply content/src/fundings/angel-match.md]: It already is!

File: content/src/quick-action-tasks/architecture-license-reinstatement.md [hunk 27]
```diff
@@ -6,34 +6,32 @@ urlSlug: architecture-license-reinstatement
...
 You need to submit a reinstatement application if:
 
-*  Your architecture license has been expired for five years or less
+- Your architecture license has been expired for five years or less
 
-:::infoAlert 
- If your license has been expired for over five years, then you need to re-apply for your architect license.
+:::infoAlert
+If your license has been expired for over five years, then you need to re-apply for your architect license.
 :::
```
Comment (adidner) [inline content/src/quick-action-tasks/architecture-license-reinstatement.md:17]: I assume that quick actions are like licenses and allowed to have no SummaryMD?

File: content/src/roadmaps/license-tasks/apply-for-shop-license.md [hunk 28]
```diff
@@ -18,12 +18,11 @@ webflowType: business-license
 webflowId: 5f77298fd2749a78af9cf7df
 licenseCertificationClassification: LICENSE
 summaryDescriptionMd: >
```
Comment (adidner) [inline content/src/roadmaps/license-tasks/apply-for-shop-license.md:20]: No >-?

File: content/src/filings/annual-report.md [hunk 29]
```diff
@@ -5,27 +5,31 @@ agency: New Jersey Division of Revenue and Enterprise Services
...
->
+- Confirmation of submission of your annual report
 
-> **What if I do not complete my report?**
->
-> After two missed annual reports, you may lose your authority to do business in New Jersey, your `Business Registration Certificate|business-reg-certificate` may be invalidated, and you could lose access to state services, including funding opportunities and licenses.
->
-> To be reinstated, businesses will need to file and pay for past reports as well as pay a reinstatement fee.
\ No newline at end of file
+:::
```
Comment (adidner) [inline content/src/filings/annual-report.md:27]: Extra ::: ?

**Review by adidner — APPROVED** [review]
Good to go. Just a few minor questions but nothing wrong that I can see.

File: content/src/quick-action-tasks/architecture-license-reinstatement.md (hunk 27 above)
Comment (seidior) [reply content/src/quick-action-tasks/architecture-license-reinstatement.md:17]: That's correct!

File: content/src/roadmaps/license-tasks/apply-for-shop-license.md (hunk 28 above)
Comment (seidior) [reply content/src/roadmaps/license-tasks/apply-for-shop-license.md:20]: Good catch. This is definitely valid (the "-" controls the behavior of multiline strings) and this exists in content that already had `summaryDescriptionMd`, but it's a good opportunity to campground.

File: content/src/filings/annual-report.md (hunk 29 above)
Comment (seidior) [reply content/src/filings/annual-report.md:27]: No, this is valid; it's for the section above. It just can't go directly beneath the text above because it's a list, and markdown lists need newlines (or `EOF`) above and below.
